src/core/Akka.Persistence.Query/EventEnvelope.cs

```diff
@@ -23,21 +24,27 @@ public sealed class EventEnvelope : IEquatable<EventEnvelope>
         /// <summary>
         /// Initializes a new instance of the <see cref="EventEnvelope"/> class.
         /// </summary>
-        [Obsolete("For binary compatibility with previous releases")]
+        [Obsolete("For binary compatibility with previous releases. Since 1.4.14")]
         public EventEnvelope(Offset offset, string persistenceId, long sequenceNr, object @event)
-            : this(offset, persistenceId, sequenceNr, @event, 0L)
+            : this(offset, persistenceId, sequenceNr, @event, 0L, null)
         { }
         
         /// <summary>
         /// Initializes a new instance of the <see cref="EventEnvelope"/> class.
         /// </summary>
+        [Obsolete("For binary compatibility with previous releases. Since 1.5.11")]
         public EventEnvelope(Offset offset, string persistenceId, long sequenceNr, object @event, long timestamp)
+            : this(offset, persistenceId, sequenceNr, @event, timestamp, null)
```

Comment:
Want to use `Array.Empty<string>` here instead of a nullable array?

Reply:
good idea